File: Plugins/Aspose_Diagram_Java_for_Ruby/lib/asposediagramjava/Layers/getalllayers.rb
module Asposediagramjava
  module GetAllLayers
    def initialize()
        data_dir = File.dirname(File.dirname(File.dirname(File.dirname(__FILE__)))) + '/data/'

        # Create instance of Diagram
        diagram = Rjb::import('com.aspose.diagram.Diagram').new(data_dir + "Drawing.vsd")

        # get Visio page
        page = diagram.getPages().getPage(0)

        layers = page.getPageSheet().getLayers()
        
        i = 0
        while i < layers.getCount()
            layer = layers.get(i)
            puts "Name: " + layer.getName().getValue().to_s
            puts "Visibility: " + layer.getVisible().getValue().to_s
            puts "Status: " + layer.getStatus().getValue().to_s
            
            i +=1
        end
    end
  end
end

File: Plugins/Aspose_Diagram_Java_for_PHP/src/aspose/diagram/WorkingwithLayers/GetAllLayers.php
<?php
namespace Aspose\Diagram\WorkingwithLayers;

use com\aspose\diagram\Diagram as Diagram;
class GetAllLayers{

    public static function run($dataDir=null){

        # Create instance of Diagram
        $diagram = new Diagram($dataDir . "Drawing.vsd");

        # get Visio page
        $page=$diagram->getPages()->getPage(0);

        $layers=$page->getPageSheet()->getLayers();

        $i = 0;
        while ($i<$layers->getCount()) {
            $layer=$layers->get($i);
            print "Name: " . (string)$layer->getName()->getValue();
            print "Visibility: " . (string)$layer->getVisible()->getValue();
            print "Status: " . (string)$layer->getStatus()->getValue();
            $i += 1;
        }
    }
}